indexing-hadoop/src/main/java/io/druid/indexer/updater/HadoopConverterJob.java

```diff
@@ -0,0 +1,1021 @@
+        context.getCounter(COUNTER_GROUP, COUNTER_LOADED)
+               .increment(size);
+      }
+      finally {
+        if (tmpDownloadFile.exists() && !tmpDownloadFile.delete()) {
+          log.warn("Temporary download file could not be deleted [%s]", tmpDownloadFile);
+        }
+      }
+      context.setStatus("CONVERTING");
+      context.progress();
+      final File outDir = new File(tmpDir, "out");
+      if (!outDir.mkdir() && (!outDir.exists() || !outDir.isDirectory())) {
+        throw new IOException(String.format("Could not create output directory [%s]", outDir));
+      }
+      IndexMaker.convert(
+          inDir,
+          outDir,
+          config.getIndexSpec(),
+          new ProgressIndicator()
+          {
+            @Override
+            public void progress()
+            {
+              context.progress();
+            }
+
+            @Override
+            public void start()
+            {
+              context.progress();
+              context.setStatus("STARTED CONVERSION");
+            }
+
+            @Override
+            public void stop()
+            {
+              context.progress();
+              context.setStatus("STOPPED CONVERSION");
+            }
+
+            @Override
+            public void startSection(String section)
+            {
+              context.progress();
+              context.setStatus(String.format("STARTED [%s]", section));
+            }
+
+            @Override
+            public void progressSection(String section, String message)
+            {
+              log.info("Progress message for section [%s] : [%s]", section, message);
+              context.progress();
+              context.setStatus(String.format("PROGRESS [%s]", section));
+            }
+
+            @Override
+            public void stopSection(String section)
+            {
+              context.progress();
+              context.setStatus(String.format("STOPPED [%s]", section));
+            }
+          }
+      );
+      final File[] files = outDir.listFiles();
+      if (files == null) {
+        throw new IOException(String.format("No files found in output directory! [%s]", outDir.getAbsolutePath()));
+      }
+      if (config.isValidate()) {
+        context.setStatus("Validating");
+        IndexIO.DefaultIndexIOHandler.validateTwoSegments(inDir, outDir);
+      }
+      context.progress();
+      context.setStatus("Starting PUSH");
+      long size = 0L;
+      for (File file : files) {
+        size += file.length();
+      }
+      final DataSegment finalSegmentOldLoadSpec = segment
+          .withVersion(
+              segment.getVersion()
+              + "_converted"
+          )
+          .withSize(size)
+          .withBinaryVersion(SegmentUtils.getVersionFromDir(outDir));
+
+      // Make the zip file
+      final DataSegment finalSegment;
+      final File tmpZipFile = File.createTempFile("zipForUpload", ".zip");
```

Comment:
we can probably reuse the code from `IndexGeneratorJob.serializeOutIndex` here to create and upload the zip file.
One nice property of that code is that it avoids the extra temporary file and writes the zip stream directly to the output target.